plugins/modules/rulebook_activation.py

```diff
@@ -518,20 +616,69 @@ def main() -> None:
             module.fail_json(msg=f"Failed to delete rulebook activation: {e}")
 
-    if activation:
-        module.exit_json(
-            msg=f"A rulebook activation with name: {name} already exists. "
-            "The module does not support modifying a rulebook activation.",
-            changed=False,
-            id=activation["id"],
-        )
-
     # Activation Data that will be sent for create/update
     activation_params = create_params(module, controller, is_aap_24=is_aap_24)
-    activation_params["name"] = name
+    activation_params["name"] = new_name if new_name else name
+
+    # Handle copies before anything else
+    if copy_from:
+        if not activation:
+            module.fail_json(msg=f"Activation with name {copy_from} was not found.")
+        else:
+            try:
+                activation_id = activation["id"]
+                copy_endpoint = f"activations/{activation_id}/copy"
+                params = {"name": name}
+
+                controller.post_endpoint(endpoint=copy_endpoint, data=params)
+                module.exit_json(changed=True)
+            except EDAError as e:
+                module.fail_json(msg=f"Failed to copy rulebook activation: {e}")
+
+    if activation:
+
+        # Define 'state' of existing activation, and remove 'enabled' from
+        # constructed parameters to avoid unnecessary updates.
+        activation["state"] = "enabled" if activation["is_enabled"] else "disabled"
+        activation_params.pop("is_enabled", None)
+        activation_params.pop("enabled", None)
+
+        # Change from list of credentials to a list of IDs in existing activation
+        credential_ids = [
+            credential_id["id"] for credential_id in activation["eda_credentials"]
+        ]
+        activation["eda_credentials"] = credential_ids
+
+        if controller.objects_could_be_different(activation, activation_params):
+            try:
+                op_type = check_operation(activation, activation_params)
+
+                # NOTE(kaiokmo): In order to avoid the user enabling/disabling and
+                # also updating an activation altogether in one shot, first we handle
+                # the operation enable/disable, then we exit. This is needed because, for now,
+                # we are not directly managing the state of an activation.
+                #
+                # This doesn't handle cases where the user tries to update an activation
+                # while it's still enabled. We simply honor the operation first before
+                # anything else.
+                if op_type != NO_OP:
+                    activation_id = activation["id"]
+                    enable_disable_endpoint = f"activations/{activation_id}/{op_type}"
+                    controller.post_endpoint(endpoint=enable_disable_endpoint)
+                    module.exit_json(changed=True)
+
+                result = controller.update_if_needed(
+                    activation,
```

Comment:
if the update is after copying, the id of activation should be updated with the new id, so the update will apply to the copied activation.